src/sinks/datadog/traces/stats.rs

```diff
@@ -0,0 +1,527 @@
+use std::collections::BTreeMap;
+
+use chrono::Utc;
+use prost::Message;
+use serde::{Deserialize, Serialize};
+use serde_bytes;
+
+use super::{ddsketch_full, sink::PartitionKey};
+use crate::{
+    event::{TraceEvent, Value},
+    metrics::AgentDDSketch,
+};
+
+const MEASURED_KEY: &str = "_dd.measured";
+const PARTIAL_VERSION_KEY: &str = "_dd.partial_version";
+const SAMPLING_RATE_KEY: &str = "_sample_rate";
+const TAG_STATUS_CODE: &str = "http.status_code";
+const TAG_SYNTHETICS: &str = "synthetics";
+const TOP_LEVEL_KEY: &str = "_top_level";
+
+#[derive(PartialEq, Eq, PartialOrd, Ord)]
+struct AggregationKey {
+    payload_key: PayloadAggregationKey,
+    bucket_key: BucketAggregationKey,
+}
+
+impl AggregationKey {
+    fn new_aggregation_from_span(
+        span: &BTreeMap<String, Value>,
+        payload_key: PayloadAggregationKey,
+        synthetics: bool,
+    ) -> Self {
+        AggregationKey {
+            payload_key: payload_key.with_span_context(span),
+            bucket_key: BucketAggregationKey {
+                service: span
+                    .get("service")
+                    .map(|v| v.to_string_lossy())
+                    .unwrap_or_default(),
+                name: span
+                    .get("name")
+                    .map(|v| v.to_string_lossy())
+                    .unwrap_or_default(),
+                resource: span
+                    .get("resource")
+                    .map(|v| v.to_string_lossy())
+                    .unwrap_or_default(),
+                ty: span
```

Comment:
Is `ty` how it's named on the DD side? Took me a minute to remember what this was when reviewing the code.

Reply:
Also no strong feeling, just had a head-scratching moment reading `ty` further down

Reply:
👀 _noticed you used `r#type` in the v2 endpoint code_ Just saying 😆